theSrc/scripts/LabeledScatter.js

```diff
@@ -80,10 +81,12 @@ class LabeledScatter {
       const plot_config = { displayModeBar: false, editable: false }
 
     const plotlyChart = await Plotly.react(this.rootElement, plot_data, plot_layout, plot_config)
-    this.drawScatterLabelLayer(plotlyChart._fullLayout, config)
+    await this.drawScatterLabelLayer(plotlyChart._fullLayout, config)
```

Comment:
Add `await` so that subsequent code can assume that the scatter label layer is drawn.